homeassistant/components/airzone/select.py

```diff
@@ -153,7 +190,18 @@ def __init__(
             f"{self._attr_unique_id}_{system_zone_id}_{description.key}"
         )
         self.entity_description = description
-        self.values_dict = {v: k for k, v in description.options_dict.items()}
+
+        if description.key == AZD_MODE:
+            options_dict: dict[str, Any] = {}
+            modes = self.get_airzone_value(AZD_MODES)
+            options_dict = {
+                k: v for k, v in description.options_dict.items() if v in modes
+            }
+            self._attr_options = list(options_dict)
+        else:
+            options_dict = description.options_dict
```

Comment:
Should I move this to a lambda function?

Reply:
I think that would be cleaner and extendable

Reply:
Done in 4fe96e4f917745209f1791cf83e762287ce9578f